simulation.cpp

```diff
@@ -396,31 +373,36 @@ void test_simulation() noexcept//!OCLINT test may be many
         identity_env.env_function_A = identity;
         assert(are_same_env_functions(identity_env.env_function_A, identity));
 
-        auto identity_net = net_param{{1,1}, linear};
+        auto identity_net_param = net_param{{1,1}, linear};
```

Comment:
rename `potential_identity_net`or something that conveys that a `network `with this `param`s _could_ behave like an `identity` function